build.sbt

```diff
@@ -1,189 +1,115 @@
-    case Some((2, minor)) if minor < 13 => true
-    case _                              => false
-  }
-
-ThisBuild / crossScalaVersions := Seq(scala212, scala213)
+ThisBuild / tlBaseVersion := "0.14"
+ThisBuild / tlCiReleaseTags := false
 
-val baseSettings = Seq(
-  scalacOptions ++= compilerOptions,
-  scalacOptions ++= (
-    if (priorTo2_13(scalaVersion.value))
-      Seq(
-        "-Xfuture",
-        "-Yno-adapted-args",
-        "-Ywarn-unused-import",
-        "-Ypartial-unification"
-      )
-    else
-      Seq(
-        "-Ymacro-annotations",
-        "-Ywarn-unused:imports"
+ThisBuild / organization := "io.circe"
+ThisBuild / crossScalaVersions := List(Scala212V, Scala213V)
+ThisBuild / scalaVersion := Scala213V
+
+ThisBuild / githubWorkflowJavaVersions := Seq("8", "11", "17").map(JavaSpec.temurin)
+
+ThisBuild / githubWorkflowAddedJobs ++= Seq(
+  WorkflowJob(
+    id = "scalafmt",
+    name = "Scalafmt",
+    scalas = List(crossScalaVersions.value.last),
+    steps = List(WorkflowStep.Checkout) ++ WorkflowStep.SetupJava(
+      List(githubWorkflowJavaVersions.value.last)
+    ) ++ githubWorkflowGeneratedCacheSteps.value ++ List(
+      WorkflowStep.Sbt(
+        List("+scalafmtCheckAll", "scalafmtSbtCheck"),
+        name = Some("Scalafmt tests")
       )
+    )
   ),
-  Compile / console / scalacOptions ~= {
-    _.filterNot(Set("-Ywarn-unused-import", "-Ywarn-unused:imports"))
-  },
-  Test / console / scalacOptions ~= {
-    _.filterNot(Set("-Ywarn-unused-import", "-Ywarn-unused:imports"))
-  },
-  coverageHighlighting := true,
-  libraryDependencies ++= Seq(
-    scalaOrganization.value % "scala-compiler" % scalaVersion.value % Provided,
-    scalaOrganization.value % "scala-reflect" % scalaVersion.value % Provided
-  ) ++ (
-    if (priorTo2_13(scalaVersion.value)) {
-      Seq(
-        compilerPlugin(("org.scalamacros" % "paradise" % paradiseVersion).cross(CrossVersion.patch))
+  WorkflowJob(
+    id = "coverage",
+    name = "Generate coverage report",
+    scalas = crossScalaVersions.value.filterNot(_.startsWith("3.")).toList,
+    steps = List(WorkflowStep.Checkout) ++ WorkflowStep.SetupJava(
+      List(githubWorkflowJavaVersions.value.last)
+    ) ++ githubWorkflowGeneratedCacheSteps.value ++ List(
+      WorkflowStep.Sbt(List("coverage", "rootJVM/test", "coverageAggregate")),
+      WorkflowStep.Use(
+        UseRef.Public(
+          "codecov",
+          "codecov-action",
+          "v2"
+        ),
+        params = Map(
+          "flags" -> List("${{matrix.scala}}", "${{matrix.java}}").mkString(",")
+        )
       )
-    } else Nil
-  ),
-  coverageEnabled := (
-    if (priorTo2_13(scalaVersion.value)) false else coverageEnabled.value
+    )
   )
 )
 
-val allSettings = baseSettings ++ publishSettings
+ThisBuild / scalafixScalaBinaryVersion := CrossVersion.binaryScalaVersion(scalaVersion.value)
+ThisBuild / scalafixAll / skip := tlIsScala3.value
+ThisBuild / ScalafixConfig / skip := tlIsScala3.value
```

Comment:
I don't think you are currently using scalafix. So there's no need for any of this.